src/core/qgseditformconfig.cpp

```diff
@@ -387,6 +388,8 @@ void QgsEditFormConfig::readXml( const QDomNode &node, QgsReadWriteContext &cont
       onRelationsLoaded();
     }
   }
+
+  context.leaveCategory();
```

Comment:
I'm thinking it might be nice to have some little raii class which automatically handles the call to leaveCategory when it gets destructed?

Reply:
This is a bit obscure to me. I spend some time reading but I fail to understand how to achieve in this case. 
You would create a shared pointer for a each call to `enterCategory` with a reference to the class?  
So if I understand correctly, every class calling should be a `QObject` (which is not the case at the moment, `QgsEditFormConfig` for instance).